contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1116,6 +1122,48 @@ datalength(PG_FUNCTION_ARGS)
 	else
 		typlen = *((int *) fcinfo->flinfo->fn_extra);
 
+	/* Handling fixed storage size datatypes. */
+	if ((*common_utility_plugin_ptr->is_tsql_tinyint_datatype)(argtypeid))
+	{
+		PG_RETURN_INT32(1);
+	}
+	else if ((*common_utility_plugin_ptr->is_tsql_smallmoney_datatype)(argtypeid) || 
+			 (*common_utility_plugin_ptr->is_tsql_smalldatetime_datatype)(argtypeid))
+	{
+		PG_RETURN_INT32(4);
+	}
+	else if (argtypeid == DATEOID)
```

Comment:
Lets put this if else ladder before typlen computation, as typlen is not required for these cases.